vnext/Microsoft.ReactNative/IReactDispatcher.h

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include "ReactDispatcherHelper.g.h"
+#include <ReactPropertyBag.h>
+#include <dispatchQueue/dispatchQueue.h>
+#include <winrt/Microsoft.ReactNative.h>
+
+namespace winrt::Microsoft::ReactNative {
+
+struct ReactDispatcher : implements<ReactDispatcher, IReactDispatcher> {
+  ReactDispatcher() = default;
+  ReactDispatcher(Mso::DispatchQueue &&queue) noexcept;
```

Comment:
>DispatchQueue [](start = 23, length = 13)

(nit) why not IDispatchQueue?

Reply:
The open source implementation of the DispatchQueue is different. We do not use the interface directly anymore. Instead, the DispatchQueue is a wrapper of the interface similar to what we do with the Mso::Future or Mso::Promise.
The good thing about having a wrapper class like Mso::DispatchQueue is that we can have template methods there, static methods, and all the DispatchQueue APIs are grouped under one class that makes it IntelliSense friendly.

---
In reply to: [423963640](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423963640) [](ancestors = 423963640)

Reply:
Does the mechanism you describe above allow people to provide their own dispatcher implementations?

---
In reply to: [423982263](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423982263) [](ancestors = 423982263,423963640)

Reply:
In the new Mso::DispatchQueue model we have only one Mso::DispatchQueue class that manages queue of tasks. Then we have IDispatchQueueScheduler interface that can run these tasks according to specific rules on the threads it wants. It is relatively small interface and developers can provide their own implementation of it. By implementing custom IDispatchQueueScheduler interface, developers can implement a 'custom queue' when they give it to the Mso::DispatchQueue constructor.

---
In reply to: [423983579](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423983579) [](ancestors = 423983579,423982263,423963640)